components/org.apache.stratos.python.cli/src/main/python/cli/CLI.py

```diff
@@ -0,0 +1,1530 @@
+        make_option('-u', '--username', type="str", help="Username of the user"),
+        make_option('-p', '--password', type="str", help="Password of the user"),
+        make_option('-f', '--json_file_path', type="str", help="Path of the JSON file")
+    ])
+    @auth
+    def do_add_cartridge(self, line , opts=None):
+        """Add a cartridge definition."""
+        try:
+            if not opts.json_file_path:
+                print("usage: add-cartridge [-f <resource path>]")
+            else:
+                cartridge = Stratos.add_cartridge(open(opts.json_file_path, 'r').read())
+                if cartridge:
+                    print("Cartridge added successfully")
+                else:
+                    print("Error adding Cartridge")
+        except BadResponseError as e:
+            self.perror(str(e))
+
+    @options([
+        make_option('-u', '--username', type="str", help="Username of the user"),
+        make_option('-p', '--password', type="str", help="Password of the user"),
+        make_option('-f', '--json_file_path', type="str", help="Path of the JSON file")
+    ])
+    @auth
+    def do_update_cartridge(self, line , opts=None):
+        """Update a cartridge"""
+        try:
+            if not opts.json_file_path:
+                print("usage: update-cartridge [-f <resource path>]")
+            else:
+                cartridge = Stratos.update_cartridge(open(opts.json_file_path, 'r').read())
+                if cartridge:
+                    print("Cartridge updated successfully")
+                else:
+                    print("Error updating Cartridge")
+        except BadResponseError as e:
+            self.perror(str(e))
+
+
+    @options([
+        make_option('-u', '--username', type="str", help="Username of the user"),
+        make_option('-p', '--password', type="str", help="Password of the user")
+    ])
+    @auth
+    def do_remove_cartridge(self, cartridge_type , opts=None):
+        """Delete a cartridge"""
+        try:
+            if not cartridge_type:
+                print("usage: remove-cartridge [cartridge-type]")
+            else:
+                cartridge_removed = Stratos.remove_cartridge(cartridge_type)
+                if cartridge_removed:
+                    print("Successfully un-deployed cartridge : "+cartridge_type)
+                else:
+                    print("Could not un-deployed cartridge : "+cartridge_type)
+        except BadResponseError as e:
+            self.perror(str(e))
+
+    """
+    # Cartridge groups
+     * list-cartridge-groups
+     * describe-cartridge-group
+     * add-cartridge-group
+     * remove-cartridge-group
+
+    """
+
+    @options([
+        make_option('-u', '--username', type="str", help="Username of the user"),
+        make_option('-p', '--password', type="str", help="Password of the user")
+    ])
+    @auth
+    def do_list_cartridge_groups(self, line , opts=None):
+        """Retrieve details of all the cartridge groups."""
```

Comment:
This Doc string is used as the help text of the command and the parameter descriptions are automatically added by the framework.  `help` attribute of the `make_option` function in `@options` decorator is used as parameter description :) 
 
![stratos-cli](https://cloud.githubusercontent.com/assets/1391468/9322605/2cfe8878-4593-11e5-8e97-b7e02e1a085b.png)